examples/source.exs

```diff
@@ -38,43 +30,14 @@ defmodule Pipeline do
       |> get_child(:muxer)
     ]
 
-    {[spec: structure], %{}}
-  end
-
-  # Once the source initializes, we grant it the control over the tcp socket
-  @impl true
-  def handle_child_notification(
-        {:socket_control_needed, _socket, _source} = notification,
-        :source,
-        _ctx,
-        state
-      ) do
-    send(self(), notification)
-
-    {[], state}
-  end
-
-  def handle_child_notification(_notification, _child, _ctx, state) do
-    {[], state}
-  end
-
-  @impl true
-  def handle_info({:socket_control_needed, socket, source} = notification, _ctx, state) do
-    case Membrane.RTMP.SourceBin.pass_control(socket, source) do
-      :ok ->
-        :ok
-
-      {:error, :not_owner} ->
-        Process.send_after(self(), notification, 200)
-    end
-
-    {[], state}
+    {[spec: structure], %{controller_pid: opts[:controller_pid]}}
   end
 
   # The rest of the module is used for self-termination of the pipeline after processing finishes
   @impl true
   def handle_element_end_of_stream(:sink, _pad, _ctx, state) do
-    {[terminate: :shutdown], state}
+    send(state.controller_pid, :eos)
+    {[], state}
```

Comment:
Usually we do `terminate: :normal` and monitor the pipeline process. This way there's no need for passing the controlling process.